### ISSUE TITLE
Feature request: add "output" to the module with "depends_on" for dependency on the module

### ISSUE THREAD

**Issue by zelig81** [terraform-aws-modules/terraform-aws-vpc#399]
Hi!
Can you add "output" to the module with "depends_on" for the possibility to depend on on the module. It will help with "count" issue that prevents from using computed values.
https://www.terraform.io/docs/configuration/outputs.html#depends_on-explicit-output-dependencies

**Comment by DrFaust92** [comment]
Hey, 

can you give an example of how would it look like?

**Comment by zelig81** [comment]
Sorry - forgot to add the example:

```terraform
output "depends_on" {
  value       = "" #no matter what it will be
  description = "output for explicit dependency on the module to wait for it"

  depends_on = [
    aws_vpc.this,
    
  ]
}

```

It will be useful in [private|database|public|intra...]_subnets

**Comment by DrFaust92** [comment]
Hey @zelig81 , is there any specific resource you need to depend on? i thought about this solution and depending on is a workaround when you can chain resource properly. im lacking the specific use case this will solve.
i can understand depending on traffic related resources like nat or igw but not subnets  for example

**Comment by zelig81** [comment]
@DrFaust92 Generally you are right, just sometimes creation of resources if not chained via `depends_on` may end in a try to create a resource that is really depends on other one that is still in the process of creation. Terragrunt solves for me that issue, but for those who prefer to use pure terraform it makes using vpc module to be problematic.

I've tried to create a working example for TGW module. In example I had to create VPCs from scratch, to interconnect it with TGW module, to attach VPNs.

**Although I'm currently not working developing the module**, but perhaps it will be still useful for people who do not use terragrunt.
